### PR TITLE
Loose coupling of pallets: Roles + Social

### PR DESCRIPTION
**Pull request by subsocialdev**

### CODE REVIEW

File: pallets/roles/src/lib.rs [hunk 1]
```diff
@@ -72,9 +84,9 @@ decl_error! {
...
     NoPermissionToManageRoles,
     /// Nothing to update in role
-    NoUpdatesInRole,
+    NoRoleUpdates,
     /// There's too many users assigned for this role to delete it
-    TooManyUserAssignedToDeleteRole,
+    TooManyUserForDeleteRole,
 
     /// No roles found for this User on specified Space
     NoAnyRolesForUserOnSpace,
```
Comment (subsocialdev) [inline pallets/roles/src/lib.rs:92]:
```suggestion
    NoRolesForUserInSpace,
```

File: pallets/roles/src/lib.rs [hunk 2]
```diff
@@ -72,9 +84,9 @@ decl_error! {
     /// Account has not permission to manage roles on this space
     NoPermissionToManageRoles,
     /// Nothing to update in role
-    NoUpdatesInRole,
+    NoRoleUpdates,
     /// There's too many users assigned for this role to delete it
-    TooManyUserAssignedToDeleteRole,
+    TooManyUserForDeleteRole,
```
Comment (subsocialdev) [inline pallets/roles/src/lib.rs:89]:
```suggestion
    TooManyUsersForDeleteRole,
```

File: pallets/roles/src/lib.rs [hunk 3]
```diff
@@ -83,15 +95,16 @@ decl_error! {
     /// No users provided when trying to grant them a Role
     NoUsersProvided,
     /// Trying to disable the role that is not enabled (or disabled as by default)
-    RoleIsNotEnabled,
+    RoleAlreadyDisabled,
```
Comment (subsocialdev) [inline pallets/roles/src/lib.rs:98]: We don't need it anymore.
```suggestion
```

File: pallets/roles/src/lib.rs [hunk 4]
```diff
@@ -83,15 +95,16 @@ decl_error! {
...
     /// Trying to enable the role that is already enabled
-    RoleIsAlreadyEnabled,
+    RoleAlreadyEnabled,
```
Comment (subsocialdev) [inline pallets/roles/src/lib.rs:100]: We don't need it anymore.
```suggestion
```

File: pallets/social/src/lib.rs [hunk 5]
```diff
@@ -372,6 +366,25 @@ decl_error! {
...
+    NoPermissionToCreatePosts,
+    /// User has no permission to create comments in this space
+    NoPermissionToCreateComments,
+    /// User has no permission to share posts/comments from this space
+    NoPermissionToShare,
+    /// User is an author, but has no permission to update own posts in this space
+    NoPermissionToUpdateOwnPosts,
+    /// User is not an author and has no permission to update posts in this space
+    NoPermissionToUpdateAnyPosts,
+    /// User has no permission to update his comments in this space posts
```
Comment (subsocialdev) [inline pallets/social/src/lib.rs:382]:
```suggestion
    /// User has no permission to update own comments in this space posts
```

File: pallets/social/src/lib.rs [hunk 6]
```diff
@@ -840,7 +872,31 @@ decl_module! {
...
       let mut post = Self::post_by_id(post_id).ok_or(Error::<T>::PostNotFound)?;
 
-      ensure!(owner == post.created.account, Error::<T>::NotAnAuthor);
+      let is_owner = owner == post.created.account;
+      let is_comment = post.is_comment();
+
+      let permission_to_check: SpacePermission;
+      let permission_error: DispatchError;
+
+      if is_comment && is_owner {
```
Comment (subsocialdev) [inline pallets/social/src/lib.rs:881]: Refactor this if/else... to such a struct:
```rust
if is_comment {
  if is_owner {
  } else {
  }
} else {
  if is_owner {
  } else {
  }
}
```

File: pallets/traits/types.json [hunk 7]
```diff
@@ -0,0 +1,6 @@
+{
+  "SpaceForRoles": {
+    "owner": "AccountId",
+    "permissions": "SpacePermissions"
```
Comment (subsocialdev) [inline pallets/traits/types.json:4]:
```suggestion
    "permissions": "Option<SpacePermissions>"
```

File: pallets/social/src/lib.rs (hunk 5 above)
Comment (subsocialdev) [inline pallets/social/src/lib.rs:382]:
```suggestion
    /// User has no permission to update own comments in this space posts
```

Comment (siman) [inline pallets/social/src/lib.rs:382]:
```suggestion
    /// User has no permission to update own comments in this space posts
```

**Pull request merged**